app/web_framework/cluster_slave_application.py

```diff
@@ -137,11 +137,23 @@ def get(self, build_id, subjob_id, atom_id):
         :type subjob_id: int
         :type atom_id: int
         """
-        max_lines = int(self.get_query_argument('maxLines', 50))
-        offset_line = self.get_query_argument('offsetLine', None)
+        max_lines = int(self.get_query_argument('max_lines', 50))
+        offset_line = self.get_query_argument('offset_line', None)
+
         if offset_line is not None:
             offset_line = int(offset_line)
-        response = self._cluster_slave.get_console_output(build_id, subjob_id, atom_id, max_lines, offset_line)
+
+        response = self._cluster_slave.get_console_output(
+            build_id,
+            subjob_id,
+            atom_id,
+            Configuration['artifact_directory'],
+            max_lines,
+            offset_line
+        )
+        # Because the 'Origin' header in the redirect (from the master) gets set to 'null', the only way
+        # for the client to receive this response is for us to allow any origin to receive this response.
+        self.set_header('Access-Control-Allow-Origin', '*')
```

Comment:
I double checked and the redirect response *does* need to have the correct allow-origin header set for the browser to follow it. This means that even if a malicious website tried to make an ajax request to our cluster master it would fail due to cross origin policy. An attack would need to be an malicious site making an ajax request to a cluster slave, which is even *less* likely to be successful since builds and subjobs are randomly distributed.

From reading around, I'm not really sure how we could do better than this. We could require the "Referer" header to match our allowed origins, but it sounds like users can legitimately disable referer headers as a security setting, so that approach would disable the redirect functionality for anyone with that security setting.